plugins/modules/zos_backup_restore.py

```diff
@@ -523,23 +540,26 @@ def restore(
         space_type (str): The unit of measurement to use when defining data set space.
         sms_storage_class (str): Specifies the storage class to use.
         sms_management_class (str): Specifies the management class to use.
+        tmp_hlq (str): : Specifies the tmp hlq to temporary datasets
     """
     args = locals()
     zoau_args = to_dunzip_args(**args)
-    response = datasets._unzip(**zoau_args)
+    output = ""
+    try:
+        rc = datasets.dunzip(**zoau_args)
+    except zoau_exceptions.ZOAUException as dunzip_exception:
+        output = dunzip_exception.response.stdout_response
+        output = output + dunzip_exception.response.stderr_response
+        rc = get_real_rc(output)
     failed = False
-    true_rc = response.rc
-    if response.rc > 0:
-        output = response.stdout_response + response.stderr_response
-        true_rc = get_real_rc(output) or true_rc
-    if true_rc > 0 and true_rc <= 4:
+    if rc > 0 and rc <= 4:
         if recover is not True:
             failed = True
-    elif true_rc > 0:
+    elif rc > 0:
```

Comment:
not a show-stopper, shouldn't this one be `elif rc>4`? Since the rc 1-4 case is covered by the if?